include/fiction/algorithms/simulation/sidb/is_operational.hpp

```diff
@@ -0,0 +1,304 @@
+//
+// Created by Jan Drewniok on 11.09.23.
+//
+
+#ifndef FICTION_IS_OPERATIONAL_HPP
+#define FICTION_IS_OPERATIONAL_HPP
+
+#include "fiction/algorithms/iter/bdl_input_iterator.hpp"
+#include "fiction/algorithms/simulation/sidb/can_positive_charges_occur.hpp"
+#include "fiction/algorithms/simulation/sidb/detect_bdl_pairs.hpp"
+#include "fiction/algorithms/simulation/sidb/exhaustive_ground_state_simulation.hpp"
+#include "fiction/algorithms/simulation/sidb/quickexact.hpp"
+#include "fiction/algorithms/simulation/sidb/quicksim.hpp"
+#include "fiction/algorithms/simulation/sidb/sidb_simulation_engine.hpp"
+#include "fiction/algorithms/simulation/sidb/sidb_simulation_parameters.hpp"
+#include "fiction/traits.hpp"
+#include "fiction/types.hpp"
+#include "fiction/utils/truth_table_utils.hpp"
+
+#include <kitty/bit_operations.hpp>
+#include <kitty/traits.hpp>
+
+#include <algorithm>
+#include <cassert>
+#include <cmath>
+#include <cstddef>
+#include <cstdint>
+#include <utility>
+#include <vector>
+
+namespace fiction
+{
+
+struct is_operational_params
+{
+    /**
+     * Possible operational status of a layout.
+     */
+    enum class operational_status
+    {
+        /**
+         * The layout is operational.
+         */
+        OPERATIONAL,
+        /**
+         * The layout is non-operational.
+         */
+        NON_OPERATIONAL
+    };
+    /**
+     * The simulation parameters for the physical simulation of the ground state.
+     */
+    sidb_simulation_parameters simulation_parameter{};
+    /**
+     * The simulation engine to be used for the operational domain computation.
+     */
+    sidb_simulation_engine sim_engine{sidb_simulation_engine::QUICKEXACT};
+    /**
+     * Parameters for the BDL pair detection algorithms.
+     */
+    detect_bdl_pairs_params bdl_params{};
+};
+
+namespace detail
+{
+/**
+ * Implementation of the `is_sp_operational` algorithm for a given gate layout.
```

Comment:
What is `is_sp_operational`?

Reply:
@marcelwa aw, sorry. Something went wrong at renaming. I will fix it.